darjeelink.gemspec

```diff
@@ -26,10 +26,27 @@ Gem::Specification.new do |s|
   s.add_dependency 'repost'
   s.add_dependency 'will_paginate'
 
+  # TODO: - MAybe we shoulf have a rubcop config for this instead?
+  #
```

Comment:
You've added `gem 'rails', '>= 6.1.7.5'` above but it's being specified already here on line 24 - see `s.add_dependency 'rails', '~> 6'`.

What's the difference between having a dependency added using `s.add_dependency` in `gemspec` and having it added in `Gemfile` 🤔 . Something to do with how these are installed as peer dependencies by darjeelink consumers?

Are we sure we're adding rails dependency config in the correct place?

Reply:
I am not 100% on the difference, it seems that this 'dev' mechanism is an old one, and deprecated.

As it is a 'dev' dependency, and I dont think it should affect things.

The portkey / darjeelink setup is over complicated IMHO. From what I can tell, this project just builds a GEM. that then has to do and be configured in portkey. That is where the actual Rails stuff is.

So 'fixing' some of these CVE's might be a bit mute IMHO. I dont feel it is worthwhile going into the minutiae of this at the moment - we have load of stuff to do - so ideally would like to upgrade, and then check portkey is still working.

Please tell me if I am wrong or misunderstood anything! I do find this stuff a bit over complex!

Reply:
This was deliberately released as a Gem so it could be open sourced.   see my comments in the slack thread linking to the original discussion.

It is different to most of our other apps and its important to follow the steps in the Readme.md.

Reply:
> What's the difference between having a dependency added using s.add_dependency in gemspec and having
>  it added in Gemfile 🤔 . Something to do with how these are installed as peer dependencies by darjeelink
> consumers?

It's because darjeelink is a Gem and not an application.

https://guides.rubygems.org/specification-reference/

Reply:
If I was to release a given Gem I'd want to be able to specify any gems it was dependant on.

So if I was releasing a gem which handled XML processing and was using Nokogiri I'd put that as a dependancy.  So its useful for things like specifying the version of ruby and gems you need to use (especially if you're using things that are only available after a certain version)

Reply:
So any change to rails version should happen in `darjeelink.gemspec` rather than to `Gemfile`, like is happening in this PR?

Reply:
I guess put another way...
This project has been working just fine having `rails` and related dependencies defined in `darjeelink.gemspec`, so what's the reason they're now being added to Gemfile?

Reply:
if the rails upgrade is to fix a security issue (which I think it is) then both should be set to ~> <rails-fixed-verison>

Reply:
I will make them consistent.

Reply:
As its a Gem anything using the Gem would use the Gemspec, but I suspect to get bundle install to work locally you need to specify them in the Gemfile.  However looking back at the code the Gemfile has never as far as I can tell specified any version numbers.

Reply:
> As its a Gem anything using the Gem would use the Gemspec, but I suspect to get bundle install to work locally you need to specify them in the Gemfile.

Yup, that sounds correct.
So
I didn't think we ever used Darjeelink in a way that wasn't a gem - i.e. we always consume it in Portkey whether that's in Prod or in Local development. So why are we now adding to Gemfile? Is there a reason?

Reply:
> As its a Gem anything using the Gem would use the Gemspec, but I suspect to get bundle install to work locally you need to specify them in the Gemfile. However looking back at the code the Gemfile has never as far as I can tell specified any version numbers.

To be honest, I cannot remember the exact details - but I imagine that is what would have happened. Was not really aware of the alt config mechanism at that point.

Reply:
I made these consistent.

Apols for lack of knowledge, I dont mind learning a but of Ruby, but if is tres annoying that all the projects seem subtly different.